tools/scripts/verify-repo-integrity.mjs

```diff
@@ -2,41 +2,135 @@
 
 import { fileURLToPath } from 'node:url';
 import { dirname, join } from 'node:path';
-import { readFileSync } from 'node:fs';
+import { readFileSync, readdirSync, statSync } from 'node:fs';
 
 import chalk from 'chalk';
 
 const __filename = fileURLToPath(import.meta.url);
 const __dirname = dirname(__filename);
+const workspaceRoot = join(__dirname, '../../');
 
 main();
 
-function main() {
-  /**
-   * @type {string[]}
-   */
-  const log = [];
+/**
+ * @typedef {{message: string; type: 'error' | 'success';}} VerifyResult
+ */
 
-  log.push(verifyPackageJson());
+function main() {
+  processVerifications(verifyYarnLock(), verifyPackageJson());
+}
 
-  log.forEach((message) => {
-    console.log(`- ✅ ${chalk.bold.green(message)}`);
+/**
+ *
+ * @param {Array<VerifyResult>} results
```

Comment:
for consistency ;)

```suggestion
 * @param {VerifyResult[]} results
```

Reply:
actually the preference should be to use generics everywhere instead of this "artificial" array syntax

Reply:
@Hotell also for the primitives? below in this code is string[], not Array<string>, so that's why suggested :)